### PR TITLE
test fix

### PR DESCRIPTION
**Pull request by dragon512**
create a directory for TS_ROOT to allow symlinks to work. This saves space on disk
Some pylint format fixes

### CODE REVIEW

File: tests/gold_tests/autest-site/trafficserver.test.ext [hunk 1]
```diff
@@ -23,261 +23,279 @@ from ports import get_port
...
+    # used as part of build. This means loctaion such as
     # PROXY_CONFIG_BIN_PATH with alway be $root/bin
-    # not something else such as bin64    
+    # not something else such as bin64
     #####
-    
+
     # set root for this test
     p.Env['TS_ROOT'] = ts_dir
+    p.MakeDir(ts_dir)
```
Comment (dragon512) [inline tests/gold_tests/autest-site/trafficserver.test.ext]: This is the real fix.. everything else is format cleanup

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/151/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/166/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/149/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1847/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/152/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1738/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/167/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/277/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/153/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/168/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1848/

**Comment by bryancall** [comment]
What is this change fixing?

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/150/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/278/

File: tests/gold_tests/autest-site/trafficserver.test.ext [hunk 2]
```diff
@@ -23,261 +23,280 @@ from ports import get_port
...
+    # used as part of build. This means loctaion such as
     # PROXY_CONFIG_BIN_PATH with alway be $root/bin
-    # not something else such as bin64    
+    # not something else such as bin64
     #####
-    
+
     # set root for this test
     p.Env['TS_ROOT'] = ts_dir
+    p.Setup.MakeDir(ts_dir)
```
Comment (dragon512) [inline tests/gold_tests/autest-site/trafficserver.test.ext]: This is the real fix

Everything else if formatting tweaks by pep8

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1739/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/410/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/279/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1849/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/411/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/412/

**Comment by zwoop** [comment]
Thanks! Fwiw, this is one of those cases where two commits would have been better, one for the indentation changes, and one for the actual fix. (But no need to redo this of course).

**Pull request merged**